### ISSUE TITLE
some confusion according to the code

### ISSUE THREAD

**Issue by saria85** [hugochan/KATE#10]
Hi, hugo,

Actually, it may not related explicitly to the issues here but I could not find an email to communicate with that. I want to apply your approach on a very large textual medical data. (Two sets of  500k documents (count of documents) in which its sizes are 400mgbyte)
If your method does well we will apply it to the different application of the medical we have.
hope you don't mind answering a couple of questions I have.
Also, I only care about having the topic modeling part so I do not have the label to calculate the accuracy.

1. why there is a couple of constructing dataset ex: 20news-group, rueters...
I know that you have checked your approach on a couple of datasets but why you make the different way of behaving with these datasets? Actually, it's confusing when there is a couple of them. 
in your example to run the code you are using this line    `python construct_20news.py -train [train_dir] -test [test_dir] -o [out_dir] -threshold [word_freq_threshold] -topn [top_n_words]`.
so when should we use the others?

2. when I run my own dataset, I would expect to have a vocabulary out of my dataset generated by the methods, then why there should be the issue of not finding some words like "interest". `(according to your saying: We can only plot word clouds for those appearing in the vocabulary. (why do we need to have an already prepare vocabulary?)  In your case, the word interest hardcoded in the original script is not part of the vocabulary. )` so what should I do if I want to visualize my own data as you did?
Also, I can see that vocabulary defined in run_lda, but not in K competitive, so why while I was running k-competitive It raises that error?

3. I want to have the same comparison you had in the paper with several approaches (I greatly appreciate your work its awesome that you have implemented almost all and include in your code and make it public :+1:  ), so I need to know the steps I need to take to apply those approach on my datasets. 
I know that it taking time to make your readme a little bit easier to follow but it will be greatly appreciated :). like you did not explain how can we resolve the issue of not finding words in the vocabulary... or running the other approaches so we can compare your approach with other approaches on very different dataset so in this case, the strongness of your approach will be even more clear as it does well on very different datasets.

Many thanks in advance  for taking time :)

**Comment by hugochan** [comment]
Hello @saria85 , 

As for your questions,
1) I used different construct_X.py files to preprocess different datasets and make them have the same data format. If you would like to try other datasets, you will need to preprocess those datasets on your own. Basically, prepare the vocabulary and Bag-of-Word representation of each document.

2) You build a vocabulary from training data and ignore all OOV words. The method learns representations for words in the vocabulary only. You can pick up some words in your own vocabulary to plot word clouds. In the word cloud plotting part of the script, I hard-coded some words when I ran the method on some financial datasets.

3) I should have made this repo more readable. (I will improve this soon!) The only work you need to do in order to apply the method as well as other baselines on your own datasets is to prepare the data (i.e., vocabulary and BOW vectors of documents). 

Hope this helps.

**Comment by saria85** [comment]
Thank you so much for your explanations :)
1. I can understand the way you preprocess the review movies may be different from 20_newsgroup. But I think 20_newsgroup and reuters are almost the same format, so I could not get why they are separate and different.

2. still Im not clear, if I create my own vocabulary, it should goes through that vocabulary and plot the representations of the word. with your saying I got that the visualization part is not consistent with the vocabulary and so if Im interested to plot, I should copy paste my words in the vocabulary somewhere else to plot them. Am I missing something? (I have not worked with word cloud sadly)

3. I will start today for preprocessing part of my work.

Thanks for taking the time:)

**Comment by hugochan** [comment]
Hello @saria85 ,

1) well, there is still some difference between them. I made separate scripts for them in order to make the code more organized. No need to worry too much about this:)

2) yes. For example, you can randomly sample some words from your vocabulary to plot the word clouds.

Good luck!
